### DISCUSSION TITLE
pseudo-residuals for zero-inflated gamma 2

### DISCUSSION THREAD

**Discussion by adeline95** [TheoMichelot/hmmTMB#20] [Q&A] [answered]
Dear authors, thank you very much for this interesting package that enables to include random effects on observation parameters. 
I am using hmmTMB version 1.0.2 on R 4.3.2 and working with activity data. I fitted a 2-states model with one response variable with distribution "zero-inflated gamma 2" and random effect on the mean of the response variable, but when trying to compute the pseudo-residuals, I got this error: 
`pr <- hmm1$pseudores()
Computing CDFs... done
Error in hmm1$pseudores() : object 'hmm' not found`

To better understand the problem, I tried to run the vignette example (energy prices) with these parameters: 

`dists <- list(Price = "zigamma2")
par0_2s <- list(Price = list(mean = c(3, 6), sd = c(1, 1), z=c(0.5,0.1)))`

But I got the same error. Therefore I assume it could be something linking to the "zigamma2" distribution. 

I hope you may be able to help me solving this problem. 

Thanks for your support and best regards, 

Adeline

**Comment by TheoMichelot** [comment, marked as answer]
Hi Adeline,

Thanks for your message, this was caused by a small bug in `HMM$pseudores()`. I fixed the bug in [commit 64aabb4](https://github.com/TheoMichelot/hmmTMB/commit/64aabb46be778546fe0271c87102e5fc7007d61d).

However, the reason this bug came up specifically in your case is that pseudo-residuals are not implemented for the zero-inflated gamma distribution. If you install the latest version of the package from Github, you will now get a warning message informing you that pseudo-residuals are not available for `"zigamma2"`. This is because it isn't straightforward to deal with observations that are exactly zero when computing pseudo-residuals.

You could consider other approaches to model checking, e.g., using simulations in `HMM$check()`.

Best wishes,
Théo

**Reply by adeline95** [reply]
Hi Theo, 
Thank you very much for your quick answer and for the heads up regarding the pseudo-residual for zero-inflated gamma distribution.
I'll explore the option of using simulations in HMM$check(). 
Thanks again for your help!
Best, 
Adeline
